reactor-core/src/test/java/reactor/test/LoggerUtils.java

```diff
@@ -1,59 +1,214 @@
 package reactor.test;
 
-import ch.qos.logback.classic.Level;
-import ch.qos.logback.classic.Logger;
-import ch.qos.logback.classic.pattern.ThrowableProxyConverter;
-import ch.qos.logback.classic.spi.ILoggingEvent;
-import ch.qos.logback.core.UnsynchronizedAppenderBase;
-import org.slf4j.LoggerFactory;
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.function.Function;
+
 import reactor.test.util.TestLogger;
+import reactor.util.Logger;
+import reactor.util.Loggers;
 
 public class LoggerUtils {
-	static final String TEST_APPENDER_NAME = "TestAppender";
+	private static Logger testLogger;
 
-	public static void addAppender(TestLogger testLogger, Class<?> classWithLogger) {
-		org.slf4j.Logger slf4jLogger = LoggerFactory.getLogger(classWithLogger);
-		if (slf4jLogger instanceof Logger) {
-			Logger logbackLogger = (Logger) slf4jLogger;
-			TestAppender appender = new TestAppender(testLogger);
-			appender.start();
-			logbackLogger.addAppender(appender);
-		}
+	public static void installAdditionalLogger(Logger testLogger) {
+		LoggerUtils.testLogger = testLogger;
 	}
 
-	public static void resetAppender(Class<?> classWithLogger) {
-		org.slf4j.Logger slf4jLogger = LoggerFactory.getLogger(classWithLogger);
-		if (slf4jLogger instanceof Logger) {
-			Logger logbackLogger = (Logger) slf4jLogger;
-			logbackLogger.detachAppender(TEST_APPENDER_NAME);
+	public static void resetAdditionalLogger() {
+		LoggerUtils.testLogger = null;
+	}
+
+	/**
+	 * Sets a {@link Loggers#useCustomLoggers(Function) logger factory} that will return loggers that not only use the
+	 * original logging framework used by reactor, but also use the logger set via {@link #installAdditionalLogger(Logger)}, irrespective
+	 * of its name or how it was obtained. The expectation here is that tests that want to assess that something is
+	 * logged by reactor will pass a {@link TestLogger} instance to {@link #installAdditionalLogger(Logger)}, trigger the operation
+	 * under scrutiny, assert the logger contents and reset state by calling {@link #resetAdditionalLogger()}.
+	 */
+	public static void setupDivertingLoggerFactory() {
```

Comment:
I'm on the fence about the method actually setting up the logger factory, vs returning a "factory" (as a `Function`) to pass to `Loggers.useCustomLoggers`.